cmake/macros/macro_check_compiler_setup.cmake

```diff
@@ -27,10 +27,16 @@
 
 MACRO(CHECK_COMPILER_SETUP _compiler_flags_unstr _linker_flags_unstr _var)
   #
+  # Strip -Wl,--as-needed to work around a serious regression in ld.bfd.
+  # See https://github.com/dealii/dealii/issues/3686
+  #
```

Comment:
Spelling things out has the advantage that the comment stays accurate even if we go through different platforms. I bet we have references to both sourceforge and google code, but none of them likely still work.